### PR TITLE
fixed local and kafka ingest that use multiple indices per ingest format and cleaned up kafka ingest and test (#436)

### PR DESCRIPTION
**Pull request by rfecher**

### CODE REVIEW

File: core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java [hunk 1]
```diff
@@ -238,88 +218,101 @@ public void consumeFromTopic(
...
-				1);
-
-		final Map<String, List<KafkaStream<byte[], byte[]>>> consumerStreams = consumer.createMessageStreams(topicCount);
-		final List<KafkaStream<byte[], byte[]>> streams = consumerStreams.get(formatPluginName);
-		int counter = 1;
-		for (final KafkaStream stream : streams) {
-			final ConsumerIterator<byte[], byte[]> it = stream.iterator();
-			queue.remove(formatPluginName);
+		try {
 			LOGGER.info("Kafka consumer setup for format [" + formatPluginName + "] against topic [" + formatPluginName + "]");
```
Comment (jprochaz) [inline core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java]: May want to set this statement back to debug, or change the comment and move it to within the batch flush statement to show ingest progress / activity.

File: core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java [hunk 2]
```diff
@@ -238,88 +218,101 @@ public void consumeFromTopic(
...
+						msg,
+						avroFormatPlugin.getAvroSchema());
+
+				if (dataRecord != null) {
+					try {
+						processMessage(
+								dataRecord,
+								ingestRunData,
+								avroFormatPlugin);
+						if (++currentBatchId > BATCH_SIZE) {
```
Comment (jprochaz) [inline core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java]: does it make sense to have BATCH_SIZE be configurable?

File: core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java [hunk 3]
```diff
@@ -42,24 +39,21 @@
 /**
  * /** This class executes the ingestion of intermediate data from a Kafka topic
  * into GeoWave.
- * 
+ *
  * @param <I>
  *            The type for the input data
  * @param <O>
  *            The type that represents each data entry being ingested
  */
```
Comment (jprochaz) [inline core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java]: update the comment to remove generic parameter type descriptions

File: core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java [hunk 4]
```diff
@@ -238,88 +218,101 @@ public void consumeFromTopic(
...
-			catch (final Exception e) {
-
-				LOGGER.warn("Consuming from Kafka topic [" + formatPluginName + "] was interrupted... ");
-
+		}
+		catch (final ConsumerTimeoutException te) {
+			if (kafkaOptions.isFlushAndReconnect()) {
+				LOGGER.info(
+						"Consumer timed out from Kafka topic [" + formatPluginName + "... ",
+						te);
```
Comment (jprochaz) [inline core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/kafka/IngestFromKafkaDriver.java]: maybe differentiate this print out to mention that a flush and reconnect will be attempted

**Comment by rfecher** [comment]
I updated it per comments on PR

**Pull request merged**